File: backend/api/models/integration_create_request.py
from typing import Dict, Any, Optional
from pydantic import BaseModel

class IntegrationCreateRequest(BaseModel):
    service_type: str
    auth_method: str
    connection_name: str
    host: str
    port: int
    database_name: str
    username: Optional[str]
    password: Optional[str]
    kerberos_principal: Optional[str]
    windows_domain: Optional[str]
    extra_options: Optional[Dict[str, Any]]

File: frontend/src/interface/integration_create_request.ts
export interface IntegrationCreateRequest {
    service_type: string;
    auth_method: string;
    connection_name: string;
    host: string;
    port: number;
    database_name: string;
    username: string | null;
    password: string | null;
    kerberos_principal: string | null;
    windows_domain: string | null;
    extra_options: string | null;
};

export interface IntegrationFormProps {
    service_type: string;
    onSubmit: (values: IntegrationCreateRequest) => void;
};